### PR TITLE
cleanup: prefer matchers over std::count_if

### PR DESCRIPTION
**Pull request by coryan**
Cleaned up most uses of `std::count_if()` in the tests in factor of a
`Contains()` or `Not(Contains())` matcher over the collection. The
matchers produce better error messages, and (maybe) better express what
we are testing for.

Thanks to @devbww for creating `ContainsOnce()` and for the review
in an earlier incarnation of this PR.

### CODE REVIEW

File: google/cloud/storage/internal/logging_resumable_upload_session_test.cc [hunk 1]
```diff
@@ -74,7 +70,7 @@ TEST_F(LoggingResumableUploadSessionTest, UploadChunk) {
   EXPECT_EQ(StatusCode::kUnavailable, result.status().code());
   EXPECT_EQ("uh oh", result.status().message());
 
-  EXPECT_EQ(1, CountLines("[UNAVAILABLE]"));
+  EXPECT_THAT(log_backend_->log_lines, Contains(HasSubstr("[UNAVAILABLE]")));
```
Comment (devbww) [inline google/cloud/storage/internal/logging_resumable_upload_session_test.cc:73]: `ContainsOnce()`?

File: google/cloud/storage/tests/object_integration_test.cc [hunk 2]
```diff
@@ -99,10 +104,8 @@ TEST_F(ObjectIntegrationTest, FullPatch) {
 
   // acl() - cannot compare for equality because many fields are updated with
   // unknown values (entity_id, etag, etc)
-  EXPECT_EQ(1, std::count_if(patched->acl().begin(), patched->acl().end(),
-                             [](ObjectAccessControl const& x) {
-                               return x.entity() == "allAuthenticatedUsers";
-                             }));
+  EXPECT_THAT(AclEntityNames(patched->acl()),
+              Contains("allAuthenticatedUsers"));
```
Comment (devbww) [inline google/cloud/storage/tests/object_integration_test.cc:108]: `ContainsOnce()`?

File: google/cloud/storage/tests/bucket_integration_test.cc [hunk 3]
```diff
@@ -689,9 +669,14 @@ TEST_F(BucketIntegrationTest, NotificationsCRUD) {
...
-  ASSERT_STATUS_OK(current_notifications);
-  EXPECT_TRUE(current_notifications->empty())
+  auto notification_ids = [&client, bucket_name] {
+    std::vector<std::string> ids;
+    auto list = client->ListNotifications(bucket_name);
+    EXPECT_STATUS_OK(list);
+    for (auto const& notification : *list) ids.push_back(notification.id());
+    return ids;
+  };
+  EXPECT_TRUE(notification_ids().empty())
```
Comment (devbww) [inline google/cloud/storage/tests/bucket_integration_test.cc:679]: The message implies that this should be `ASSERT_TRUE`.

File: google/cloud/storage/tests/object_integration_test.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/tests/object_integration_test.cc]: Fixed

File: google/cloud/storage/internal/logging_resumable_upload_session_test.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/internal/logging_resumable_upload_session_test.cc]: Fixed

File: google/cloud/storage/tests/bucket_integration_test.cc (hunk 3 above)
Comment (coryan) [reply google/cloud/storage/tests/bucket_integration_test.cc]: Fixed, also searched for any other "Test aborted" and found more uses of `std::count_if()` which were easier to cleanup thanks to your new matcher.

**Comment by coryan** [comment]
Rebased and fixed the issues @devbww brought up. PTAL.

**Pull request merged**